### PR TITLE
[DLS-5176] New layout file using play-frontend-hmrc library

### PR DESCRIPTION
**Pull request by ruthdas03**
Changes the layout file and pages using it:

- Eligibility page
- Confirm email page
- Select email page
- Email savings reminder page

### CODE REVIEW

**Comment by ruthdas03** [comment]
This pr includes changes for Layout file and the below pages using it
Eligibility page
Confirm email page
Select email page
Email savings reminder page

File: app/uk/gov/hmrc/helptosavefrontend/views/register/you_are_eligible.scala.html [hunk 1]
```diff
@@ -47,17 +48,11 @@ <h1 class="govuk-heading-xl">
       <li>@Messages("hts.register.you-are-eligible.li2")</li>
     </ul>
 
-    @form(routes.EligibilityCheckController.youAreEligibleSubmit) {
-        @CSRF.formField(request)
-        @submit('_buttonClass -> "", 'id → "start-creating-account") {
-            @Messages("hts.global.continue-button")
-        }
-    }
```
Comment (LAJW) [inline app/uk/gov/hmrc/helptosavefrontend/views/register/you_are_eligible.scala.html:56]: Why was this form removed?

File: app/uk/gov/hmrc/helptosavefrontend/views/register/you_are_eligible.scala.html [hunk 2]
```diff
@@ -47,17 +48,11 @@ <h1 class="govuk-heading-xl">
...
 <div class="govuk-form">
     @viewHelpers.form(routes.EligibilityCheckController.youAreEligibleSubmit) {
-    <!--@CSRF.formField(request)-->
```
Comment (LAJW) [inline app/uk/gov/hmrc/helptosavefrontend/views/register/you_are_eligible.scala.html]: Why was this code commented out?

File: app/uk/gov/hmrc/helptosavefrontend/views/email/select_email.scala.html [hunk 3]
```diff
@@ -59,41 +56,38 @@ <h2 class="heading-medium error-summary-heading" id="error-summary-invalid-email
...
+          <div class="@{"govuk-radios__conditional govuk-radios__conditional--hidden" + {if(hasInvalidEmail) " govuk-panel-with-error" else ""}}" id="collect-new-email">
+              <div class="@{"govuk-form-group" + {if(hasInvalidEmail) " govuk-form-group-error" else ""}}">
+                <label class="govuk-label" for="new-email"><strong>@messages("hts.select-email.enter-new-address-label")</strong></label>
+                <span class="govuk-hint" id="new-email-hint">@Html(Messages("hts.select-email.enter-new-address-hint"))</span>
+                @if(hasInvalidEmail) { <span class="govuk-error-message" id="new-email-error">@email_form_validation_error_message(form, "new-email")</span> }
+                <input class="govuk-input govuk-!-width-one-half @if(hasInvalidEmail){form-control-error}" name="new-email" type="email" id="new-email" aria-describedby="new-email-hint @if(hasInvalidEmail || hasNotSelected){ new-email-error}" @if(form.data.get("new-email").nonEmpty){ value="@form.data.get("new-email")" }>
+              </div>
+            <p class="govuk-label">@messages("hts.select-email.enter-new-address-body-text")</p>
+         </div>
+      </div>
```
Comment (LAJW) [inline app/uk/gov/hmrc/helptosavefrontend/views/email/select_email.scala.html:85]: ⛏️ I wish more substantial changes like this were separated into its own commit.

**Review by LAJW — COMMENTED** [review]
I'd like some questions answered.

File: conf/app.routes [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # microservice specific routes
-
+->          /hmrc-frontend                                         hmrcfrontend.Routes
```
[the remark this answers is not in the document]
Comment (ruthdas03) [reply conf/app.routes:2]: So this route is added inorder to use the HMRC frontend library, one of the [frontend migration step](https://confluence.tools.tax.service.gov.uk/pages/viewpage.action?spaceKey=DDCWLS&title=Frontend+Migration+Process)

File: app/uk/gov/hmrc/helptosavefrontend/views/email/select_email.scala.html (hunk 3 above)
Comment (ruthdas03) [reply app/uk/gov/hmrc/helptosavefrontend/views/email/select_email.scala.html:85]: So do you want me to revert these changes and do a new pr for the same.

Comment (LAJW) [reply app/uk/gov/hmrc/helptosavefrontend/views/email/select_email.scala.html:85]: No need to create a separate PR, just _commit_ (you could do an interactive rebase + edit/partial staging and split this huge commit into two).

That's just a wish though, I've looked at it, and it seems OK. It'd just be easier to review (commit title is a bit misleading). Keep it in mind next time.

File: app/uk/gov/hmrc/helptosavefrontend/views/register/you_are_eligible.scala.html [hunk 5]
```diff
@@ -52,5 +53,13 @@ <h1 class="heading-large">
             @Messages("hts.global.continue-button")
         }
     }
+<div class="govuk-form">
+    @viewHelpers.form(routes.EligibilityCheckController.youAreEligibleSubmit) {
+    <!--@CSRF.formField(request)-->
```
[the remark this answers is not in the document]
Comment (ruthdas03) [reply app/uk/gov/hmrc/helptosavefrontend/views/register/you_are_eligible.scala.html]: It seems it [uncommented](https://github.com/hmrc/help-to-save-frontend/blob/327f0af9522b4c433e70abbf2edc0d49facc5964/app/uk/gov/hmrc/helptosavefrontend/views/register/you_are_eligible.scala.html#L51-L55) (unless I'm confused here 😃 )

File: app/uk/gov/hmrc/helptosavefrontend/views/register/you_are_eligible.scala.html (hunk 1 above)
Comment (ruthdas03) [reply app/uk/gov/hmrc/helptosavefrontend/views/register/you_are_eligible.scala.html:56]: Now that we are using the govuk form, it's been replaced.

File: app/uk/gov/hmrc/helptosavefrontend/views/layout.scala.html [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    @contentBlock
+    @hmrcReportTechnicalIssueHelper()
+}
+@pageTitle = @{
+    title + " - " + Messages("app.name") + " - " + "GOV.UK"
+}
+@hmrcLayout(
+    pageTitle = Some(pageTitle),
+    isWelshTranslationAvailable = false, /* or false if your service has not been translated */
+    //signOutUrl = Some(routes.SignOutController.signout.url),
```
[the remark this answers is not in the document]
Comment (ruthdas03) [reply app/uk/gov/hmrc/helptosavefrontend/views/layout.scala.html]: Removed it.

**Review by LAJW — DISMISSED** [review]

**Comment by nubz** [comment]
We are closing this PR in favour of using this branch to generate smaller PRs

**Pull request closed without merging**